napari/utils/perf/_patcher.py

```diff
@@ -0,0 +1,155 @@
+    except AttributeError:
+        raise PatchError(
+            f"Parent {parent_str} has no attribute {callable_str}"
+        )
+
+    label = (
+        callable_str if class_str is None else f"{class_str}.{callable_str}"
+    )
+
+    # Patch it with the user-provided patch_func.
+    print(f"Patcher: patching {module.__name__}.{label}")
+    patch_func(parent, callable_str, label)
+
+
+def _import_module(target_str: str) -> Tuple[types.ModuleType, str]:
+    """Import the module portion of this target string.
+
+    Try importing successively longer segments of the target_str. For example
+    with "napari.utils.chunk_loader.ChunkLoader.load_chunk" we will import:
+        napari (success)
+        napari.utils (success)
+        napari.utils.chunk_loader (success)
+        napari.utils.chunk_loader.ChunkLoader (failure)
+
+    The last one fails because ChunkLoader is a class not a module.
+
+    Returns
+    -------
+    Tuple[types.ModuleType, str]
+        Where the module is the inner most imported module, and the string
+    is the rest of target_str that was not modules.
+    """
+    parts = target_str.split('.')
+    module = None  # Inner-most module imported so far.
+
+    # Progressively try to import longer and longer segments of the path.
+    for i in range(1, len(target_str)):
+        module_path = '.'.join(parts[:i])
+        try:
+            module = import_module(module_path)
+        except ModuleNotFoundError:
+            if module is None:
+                # The very first top-level module import failed!
+                raise PatchError(f"Module not found: {module_path}")
+
+            # We successfully imported part of the target_str but then
+            # we got a failure. Usually this is because we tried
+            # importing a class or function. Return the inner-most
+            # module we did successfuly import. And return the rest of
+            # the module_path we didn't use.
+            attribute_str = '.'.join(parts[i - 1 :])
+            return module, attribute_str
+
+
+def patch_callables(callables: List[str], patch_func: PatchFunction) -> None:
+    """Patch the given list of callables.
+
+    Parameters
+    ----------
+    callables : List[str]
+        Patch all of these callables (functions or methods).
+    patch_func : PatchFunction
+        Called on every callable to patch it.
+
+    Notes
+    -----
+    The callables list should look like:
+    [
+        "module.module.ClassName.method_name",
+        "module.function_name"
+        ...
+    ]
+
+    Nested classes and methods not allowed, but support could be added.
+
+    An example patch_func is:
+
+    import wrapt
+    def _my_patcher(parent: CallableParent, callable: str, label: str) -> None:
+        @wrapt.patch_function_wrapper(parent, callable)
+        def my_announcer(wrapped, instance, args, kwargs):
+            print(f"Announce {label}")
+            return wrapped(*args, **kwargs)
```

Comment:
This won't render properly as is. For reference, here's a cheat sheet for ReStructuredText syntax, which is what's being used in these docstrings: https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#inserting-code-and-literal-blocks

```suggestion
    An example patch_func is::

        import wrapt
        def _my_patcher(parent: CallableParent, callable: str, label: str):
            @wrapt.patch_function_wrapper(parent, callable)
            def my_announcer(wrapped, instance, args, kwargs):
                print(f"Announce {label}")
                return wrapped(*args, **kwargs)
```